app/src/main/java/dev/ukanth/ufirewall/log/LogRecyclerViewAdapter.java

```diff
@@ -107,42 +98,37 @@ private Bitmap getAppIcon26(PackageManager mPackageManager, ApplicationInfo appl
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        data = logData.get(position);
+        LogData data = logData.get(position);
         PackageManager manager = context.getPackageManager();
-        holder.bind(logData.get(position),recyclerItemClickListener);
+        holder.bind(logData.get(position), recyclerItemClickListener);
         try {
-            HashMap<Integer,String> listMaps = Api.getPackagesForUser(Api.getListOfUids());
-            info = Api.getPackageDetails(context, listMaps, data.getUid());
-            if(info!=null && info.applicationInfo !=null) {
-                Object drawable = info.applicationInfo.loadIcon(manager);
-                if (drawable instanceof Bitmap)
-                    holder.icon.setImageBitmap((Bitmap)drawable);
-                else
-                    holder.icon.setBackground((Drawable)drawable);
-            } else{
+            HashMap<Integer, String> listMaps = Api.getPackagesForUser(Api.getListOfUids());
+            PackageInfo info = Api.getPackageDetails(context, listMaps, data.getUid());
+            if (info != null && info.applicationInfo != null) {
+                Drawable drawable = info.applicationInfo.loadIcon(manager);
+                holder.icon.setBackground(drawable);
+            } else {
                 Drawable appIcon = context.getDrawable(R.drawable.ic_unknown);
                 holder.icon.setImageBitmap(Api.getBitmapFromDrawable(appIcon));
```

Comment:
The
`holder.icon.setBackground(drawable);`
is not consistent with the similar code a few lines below:
```
if (Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.O) {
    holder.icon.setBackground(context.getDrawable(R.drawable.ic_unknown));
} else {
    Drawable appIcon = context.getDrawable(R.drawable.ic_unknown);
    holder.icon.setImageBitmap(Api.getBitmapFromDrawable(appIcon));
}
```
At first glance it would seem that
`holder.icon.setImageBitmap(Api.getBitmapFromDrawable(drawable))`
should be used at that spot instead.